pkg/apis/steward/v1alpha1/run_types.go

```diff
@@ -74,6 +76,15 @@ type PipelineSpec struct {
 	RunDetails *PipelineRunDetails `json:"runDetails,omitempty"`
 }
 
+// ImageSpec is the definition of an image with an oprional pull policy
+type ImageSpec struct {
+	// Image is the image name with the tag
+	Image string `json:"image,omitempty"`
+
+	// PullPolicy is the pull policy for the image
+	PullPolicy string `json:pullPolicy,omitempty`
+}
+
```

Comment:
Having a top-level element only for the JFR image aspect is hampers future extensions. Why not creating a spec subsection to configure _anything_ about the Jenkinsfile Runner, not just the image?

It would look like this (with some fantasy additions to illustrate future extensions):

```yaml
spec:
    ...
    jenkinsfileRunner:
        image: image1
        imagePullPolicy: Always
        resources:
            requests:
                cpu: 5
                memory: 1.7GiB
        enableFeatureX: true
        enableFeatureY: false
```

Reply:
I think the jfr section in `spec` looks like it is made not only for `image` configuration but can be extended. The pipeline spec parameter [JenkinsfileRunnerImage of type ImageSpec](https://github.com/SAP/stewardci-core/blob/4997c286dda4699b793e820f625b9d59e1ec97cf/pkg/apis/steward/v1alpha1/run_types.go#L36) is extensible because of [type ImageSpec](https://github.com/SAP/stewardci-core/blob/4997c286dda4699b793e820f625b9d59e1ec97cf/pkg/apis/steward/v1alpha1/run_types.go#L80). It may need cosmetic renaming to make more abstract.

Reply:
Changed